### PR TITLE
Add support for authentication using bearer tokens

### PR DESCRIPTION
**Pull request by scb01**
Tested by running tests locally.

### CODE REVIEW

**Review by leeyun53 — APPROVED** [review]
Looks good to me!

File: tests/integration_test.rs [hunk 1]
```diff
@@ -45,27 +46,44 @@ mod mocks;
 async fn end_to_end_test() {
     // Setup mock otlp servers for filtered logs
     let self_signed_cert = generate_self_signed_cert();
-
-    let filtered_target = MockServer::new(free_local_port_in_range(10000..=10100).unwrap(), None);
+    let auth_enabled = true;
+    let auth_disabled = false;
```
Comment (nlcamp) [inline tests/integration_test.rs]: nit: probably better to define these as `const`s at the top of the file so they aren't mistakenly assigned the wrong values by future maintainers.

File: tests/integration_test.rs [hunk 2]
```diff
@@ -413,3 +437,8 @@ async fn validate_unfiltered_logs(
         .await
         .is_err());
 }
+
+// Dummy method that just passes in an UUID as a bearer token
+fn get_test_bearer_token() -> String {
+    format!("{}", Uuid::new_v4())
```
Comment (nlcamp) [inline tests/integration_test.rs]: Consider writing it to a tempfile so the mock server can also read it and validate the incoming header.

**Review by nlcamp — APPROVED** [review]

File: tests/integration_test.rs (hunk 1 above)
Comment (scb01) [reply tests/integration_test.rs]: Good idea. Done.

File: tests/integration_test.rs (hunk 2 above)
Comment (scb01) [reply tests/integration_test.rs]: Done. Added to temp file and validated in mock server.

**Pull request merged**